### ISSUE TITLE
get_task_settings():  subtask settings not read properly for Scan

### ISSUE THREAD

**Issue by pmendes** [copasi/basico#57]
I have a file with a parameter scan that has no tick on "Sub Task Output" (ie 'before', 'during' and 'after' are all unticked).

Then I read the file with BasiCO with this code:

```
mymodel = load_model("test.cps")
task = get_task_settings('Scan')
print(task)
```
and it shows:
```
'problem': {'Subtask': 0, 'Output in subtask': True, 'Adjust initial conditions': False, 'Continue on Error': False, 'Subtask Output': 'none'}
```

then I change the filename of the report and save the model to a new file:

```
oldfilename = task['report']['filename']
task['report']['filename'] = '0_' + oldfilename
set_task_settings('Scan', task)
save_model("test_new.cps")
```

When I load this new model into CopasiUI it now shows the sub task output as "during"  (instead of nothing ticked), as a consequence there is no output.

In summary: original file has no tick on sub task output, I change the report filename with BasiCO, then the new file has a tick on Sub task output "during".

**Comment by pmendes** [comment]
I just realized that my issue title may be incorrect, I am not sure if BasiCO is making a mistake reading or writing...

**Comment by fbergmann** [comment]
It might be the newest version of the language bindings. Where we have 3 options for when to collect data for the task. If you updated `python-copasi`, then there could be changes needed. I'll investigate first thing Monday.

**Comment by fbergmann** [comment]
I've now added support so you can just specify the output specification when using set_scan_settings. Supported are arrays ['Before', 'During', 'After'] ... note that the behavior of newer COPASI is that only the `Subtask Output` field is read. There it is specified in your output as 'none' so no output would have been collected. 

I assume the inconsistency comes that you had a newer version of CopasiUI with an older version of `python-copasi`.  I'd suggest to update to the latest python copasi one doing: `pip install --upgrade python-copasi`

**Comment by pmendes** [comment]
Ok, will try that. I noticed that python-copasi is still on version 4.42 ?

**Comment by fbergmann** [comment]
not for python 3.9+ ... are you still using older versions of python?

**Comment by pmendes** [comment]
on the specific machine that runs cloud-copasi python 3.8, on my own desktop 3.9

**Comment by pmendes** [comment]
Frank, in CopasiUI when you have none ticked, you just don't get output from the subtask but you still get output from the scan task itself (if you specify a report there). This is how I have to get output from Steady State: create the report template where the steady state concentrations are output, assign this template to a report in scan and then remove the ticks on during, before, after (ie none). Otherwise I can't get output

**Comment by fbergmann** [comment]
indeed .. the problem is just that the python-copasi version writes a different COPASI file than the new UI reads. In the old version when you had "Output in subtask": False .. that would be equivalent with the new option 'Subtask Output': None ... I'm not sure why in your case you have the first one set to True. 

I'll create 3.8 binaries for 4.44 for linux x64 tomorrow.

**Comment by fbergmann** [comment]
I've uploaded a 3.7 and 3.8 (and source archive) to pypi, with that it should be possible to install 4.44 on all systems. It might still be a good idea to update your cloud-copasi distribution at some point, since 3.8 is nearing its end of life.
